eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -544,3 +544,52 @@ function CheckArtifactShaAgainstTagsList($priorExistingTagList, $releaseSha, $ap
     exit(1)
   }
 }
+
+function Get-CSVMetadata ([string]$MetadataUri) {
+  $metadataResponse = Invoke-RestMethod -Uri $MetadataUri -method "GET" -MaximumRetryCount 3 -RetryIntervalSec 10 | ConvertFrom-Csv
+  return $metadataResponse
+}
+
+function Get-BlobStorage-Artifacts($blobStorageUrl, $blobDirectoryRegex) {
+  LogDebug "Reading artifact from storage blob ..."
+  $returnedArtifacts = @()
+  $pageToken = ""
+  Do {
+      $resp = ""
+      if (!$pageToken) {
+          # First page call.
+          $resp = Invoke-RestMethod -Method Get -Uri $blobStorageUrl
+      }
+      else {
+          # Next page call
+          $blobStorageUrlPageToken = $blobStorageUrl + "&marker=$pageToken"
+          $resp = Invoke-RestMethod -Method Get -Uri $blobStorageUrlPageToken
+      }
+      # Convert to xml documents. 
+      $xmlDoc = [xml](removeBomFromString $resp)
+      foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+          # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+          $artifact = $elem.Name -replace $blobDirectoryRegex, '$1'
+          $returnedArtifacts += $artifact
+      }
+      # Fetch page token
+      $pageToken = $xmlDoc.EnumerationResults.NextMarker
+  } while ($pageToken)
+  return $returnedArtifacts
+}
+
+# The sequence of Bom bytes differs by different encoding. 
+# The helper function here is only to strip the utf-8 encoding system as it is used by blob storage list api.
+# Return the original string if not in BOM utf-8 sequence.
+function removeBomFromString([string]$bomAwareString) {
```

Comment:
I know you didn't add this function but do you know why it is needed? If we need to keep it we should fix the casing "RemoveBOMFromString"

Reply:
Storage response body use BOM UTF-8 encoding which we have to parse it first before reading the content.
https://en.wikipedia.org/wiki/Byte_order_mark

We have to use the helper function. Java has the same helper function to trim the body.

I treated kind of like internal helper method. That's why I use lower case in the beginning. I can make it Capitalize the method name.